images/airflow/2.9.2/python/mwaa/celery/task_monitor.py

```diff
@@ -273,25 +291,62 @@ def _cleanup_undead_process(process_id: int):
     )
 
 
+def _get_next_unprocessed_signal() -> (str, dict):
```

Comment:
What happens if a signal is sent, then 30 minutes later container exits due to some unrelated issue and is replaced. Since sidecar syncs anything in past hour, will this new container re-process this old signal. Do we need a mechanism preventing signals from being processed that predate container launch?

Reply:
When the container exits, the corresponding Fargate task will be terminated and replaced with a new one with its own Fargate task ID and a new EBS volume. There is no data from the old Fargate task/container in the EBS volume that could get processed by the new containers. And when the serviceEni container lookup for data in the EFS volume, it only takes data corresponding to the current Fargate task, so again any old data for the previous task is not picked up.